extensions/amp-story/1.0/pagination-buttons.js

```diff
@@ -329,8 +330,13 @@ export class PaginationButtons {
 
     if (pageIndex === totalPages - 1) {
       this.ampStory_.hasBookend().then((hasBookend) => {
+        const viewer = Services.viewerForDoc(this.ampStory_.element);
         if (!hasBookend) {
```

Comment:
By default viewer behavior, we never show the bookend when the story is embedded in a player (see lines here in `amp-story.js`):

https://github.com/ampproject/amphtml/blob/36b8196c6d7b50ca3c5547471cbaafa3523123d5/extensions/amp-story/1.0/amp-story.js#L424-L427

so we can remove one level of `if()`:
```
      this.ampStory_.hasBookend().then((hasBookend) => {
        const viewer = Services.viewerForDoc(this.ampStory_.element);
        if (viewer.hasCapability('swipe')) {
          this.forwardButton_.updateState(ForwardButtonStates.NEXT_PAGE);
        } else if (!hasBookend) {
          this.forwardButton_.updateState(ForwardButtonStates.REPLAY);
        }
      });
```

Reply:
Not a big deal, but making assumptions based on business logic from another file is always fragile and easy to break.
Basing this code on whether there is a bookend will always be more robust than basing it on the conditions for which there's a bookend, which could change at any time.

Reply:
Hm yeah I looked at the logic and I see what you're saying since `viewer.hasCapability('swipe')` implies `!hasBookend`. That said, I think I agree with Gabriel; it seems inexpensive to just do the double check in case that logic ever changes.